fontc/src/lib.rs

```diff
@@ -2028,4 +2053,211 @@ mod tests {
+        // VarStore (without a mapping) can be built. In this particular case, this
+        // turns out to be more compact than the equivalent indirect VarStore
+        // so we check it gets preferred over the latter.
+        let temp_dir = tempdir().unwrap();
+        let build_dir = temp_dir.path();
+        compile(Args::for_test(
+            build_dir,
+            "HVAR/SingleModel_Direct/HVARSingleModelDirect.designspace",
+        ));
+        let font_file = build_dir.join("font.ttf");
+        let buf = fs::read(font_file).unwrap();
+        let font = FontRef::new(&buf).unwrap();
+        let num_glyphs = font.maxp().unwrap().num_glyphs();
+        assert_eq!(num_glyphs, 14);
+        let hvar = font.hvar().unwrap();
+        assert!(hvar.advance_width_mapping().is_none());
+        let varstore = hvar.item_variation_store().unwrap();
+        let regions = varstore
+            .variation_region_list()
+            .unwrap()
+            .variation_regions();
+        // this simple test font only has two masters (Regular [default] and Bold)
+        // so we expect a single region with a single wght axis
+        assert_eq!(regions.len(), 1);
+        let region_coords = regions
+            .get(0)
+            .unwrap()
+            .region_axes()
+            .iter()
+            .map(|coords| {
+                [
+                    coords.start_coord().to_f32(),
+                    coords.peak_coord().to_f32(),
+                    coords.end_coord().to_f32(),
+                ]
+            })
+            .collect::<Vec<_>>();
+        assert_eq!(region_coords, vec![[0.0, 1.0, 1.0]]);
+        // we expect one ItemVariationData and one delta set per glyph
+        assert_eq!(varstore.item_variation_data_count(), 1);
+        let vardata = varstore.item_variation_data().get(0).unwrap().unwrap();
+        assert_eq!(vardata.region_indexes(), &[0]);
+        assert_eq!(vardata.item_count(), num_glyphs);
+        assert_eq!(vardata.delta_set(0).collect::<Vec<_>>(), vec![0]);
+        assert_eq!(vardata.delta_set(1).collect::<Vec<_>>(), vec![100]);
+        assert_eq!(vardata.delta_set(2).collect::<Vec<_>>(), vec![120]);
+        assert_eq!(vardata.delta_set(3).collect::<Vec<_>>(), vec![70]);
+        assert_eq!(vardata.delta_set(4).collect::<Vec<_>>(), vec![100]);
+        assert_eq!(vardata.delta_set(5).collect::<Vec<_>>(), vec![120]);
+        assert_eq!(vardata.delta_set(6).collect::<Vec<_>>(), vec![120]);
+        assert_eq!(vardata.delta_set(7).collect::<Vec<_>>(), vec![70]);
+        assert_eq!(vardata.delta_set(8).collect::<Vec<_>>(), vec![70]);
+        assert_eq!(vardata.delta_set(9).collect::<Vec<_>>(), vec![120]);
+        assert_eq!(vardata.delta_set(10).collect::<Vec<_>>(), vec![70]);
+        assert_eq!(vardata.delta_set(11).collect::<Vec<_>>(), vec![0]);
+        assert_eq!(vardata.delta_set(12).collect::<Vec<_>>(), vec![0]);
+        assert_eq!(vardata.delta_set(13).collect::<Vec<_>>(), vec![0]);
```

Comment:
Optional, I find it easier to read as shown in https://github.com/googlefonts/fontc/pull/525